rust_snuba/src/processors/spans.rs

```diff
@@ -0,0 +1,406 @@
+            "system".into(),
+            "transaction".into(),
+            "transaction.method".into(),
+            "transaction.op".into(),
+        ]
+    }
+
+    fn to_values(&self) -> Vec<Value> {
+        vec![
+            self.action.clone().into(),
+            self.domain.clone().into(),
+            self.group.clone().into(),
+            self.http_method.clone().into(),
+            self.module.clone().into(),
+            self.op.clone().into(),
+            self.status.clone().as_str().into(),
+            self.status_code.unwrap_or(0).to_string().into(),
+            self.system.clone().into(),
+            self.transaction.clone().into(),
+            self.transaction_method.clone().into(),
+            self.transaction_op.clone().into(),
+        ]
+    }
+}
+
+#[derive(Debug, Default, Serialize)]
+struct Span {
+    deleted: u8,
+    description: String,
+    duration: u64,
+    end_ms: u64,
+    end_timestamp: u64,
+    exclusive_time: u64,
+    group: u64,
+    group_raw: u64,
+    is_segment: bool,
+    #[serde(rename(serialize = "measurements.key"))]
+    measurement_keys: Vec<String>,
+    #[serde(rename(serialize = "measurements.value"))]
+    measurement_values: Vec<Value>,
+    offset: u64,
+    parent_span_id: u64,
+    partition: u16,
+    platform: String,
+    project_id: u64,
+    retention_days: u32,
+    segment_id: u64,
+    segment_name: String,
+    #[serde(flatten)]
+    sentry_tags: SentryTags,
+    #[serde(rename(serialize = "sentry_tags.key"))]
+    sentry_tag_keys: Vec<String>,
+    #[serde(rename(serialize = "sentry_tags.value"))]
+    sentry_tag_values: Vec<Value>,
+    span_id: u64,
+    span_kind: String,
+    span_status: u8,
+    start_ms: u64,
+    start_timestamp: u64,
+    status: u8,
+    #[serde(rename(serialize = "tags.key"))]
+    tag_keys: Vec<String>,
+    #[serde(rename(serialize = "tags.value"))]
+    tag_values: Vec<Value>,
+    trace_id: Uuid,
+    transaction_id: Uuid,
+    transaction_op: String,
+    user: String,
+}
+
+impl TryFrom<FromSpanMessage> for Span {
+    type Error = InvalidMessage;
+    fn try_from(from: FromSpanMessage) -> Result<Span, InvalidMessage> {
+        let end_timestamp_ms = from.start_timestamp_ms + from.duration_ms;
+        let status = from.sentry_tags.status as u8;
+        let transaction_op = from.sentry_tags.transaction_op.clone();
+        let mut tag_keys: Vec<String> = from.tags.clone().into_keys().collect();
+        let mut tag_values: Vec<Value> = from
+            .tags
+            .into_values()
+            .map(|v| match v {
+                Value::String(v) => Value::String(v),
+                Value::Bool(b) => {
+                    let mut v: Vec<char> = b.to_string().as_str().chars().collect();
+                    v[0] = v[0].to_uppercase().nth(0).unwrap();
+                    Value::String(v.into_iter().collect())
+                }
+                _ => Value::String(v.to_string()),
+            })
+            .collect();
```

Comment:
This is horrible but it's needed to match the exact Python output as it's making everything a string (so a `bool` becomes `True` where it's would be `true` if transforming the JSON equivalent to a string).

Ideas to improving this situation are welcome.

Reply:
It's easier for my brain as:
```
match b {
  True => "True",
  False => "False"
}
```

Reply:
Aside, should this conversion be happening in Snuba at all? Tag values are only supposed to be strings, shouldn't we be checking that somewhere upstream of here? In datasets where data is written to nodestore or other storages, it should be consistent everywhere.

Reply:
Good point. https://github.com/getsentry/sentry/pull/56362